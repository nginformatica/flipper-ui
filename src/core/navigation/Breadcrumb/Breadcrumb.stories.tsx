import React from 'react'
import { Meta, StoryFn } from '@storybook/react'
import Link from '@/core/navigation/Link'
import Breadcrumb from '.'

export default {
    title: 'Navigation/Breadcrumb',
    component: Breadcrumb,
    argTypes: {
        backgroundColor: { control: 'color' }
    }
} as Meta<typeof Breadcrumb>

const Template: StoryFn<typeof Breadcrumb> = args => <Breadcrumb {...args} />

export const Default = Template.bind({})
Default.args = {
    children: (
        <Breadcrumb>
            <Link color='inherit' href='#'>
                Flipper-ui
            </Link>
            <Link color='inherit' href='#'>
                Breadcrumb
            </Link>
        </Breadcrumb>
    )
}
