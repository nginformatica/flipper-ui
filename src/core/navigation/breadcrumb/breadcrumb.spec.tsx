import React from 'react'
import { render, screen } from '@testing-library/react'
import Link from '@/core/navigation/link'
import Breadcrumb from '.'

describe('Drawer', () => {
    it('should render', () => {
        render(<Breadcrumb data-testid='breadcrumb' />)

        const breadcrumb = screen.getByTestId('breadcrumb')

        expect(breadcrumb).toBeDefined()
    })

    it('should render children', () => {
        render(
            <Breadcrumb data-testid='breadcrumb'>
                <Link color='inherit' href='#'>
                    Flipper-ui
                </Link>
                <Link color='inherit' href='#'>
                    Breadcrumb
                </Link>
            </Breadcrumb>
        )

        const firstChildren = screen.getByText('Flipper-ui')
        const secondChildren = screen.getByText('Breadcrumb')

        expect(firstChildren).toBeDefined()
        expect(secondChildren).toBeDefined()
    })

    it('should match snapshot', () => {
        const { container } = render(
            <Breadcrumb>
                <Link color='inherit' href='#'>
                    Flipper-ui
                </Link>
                <Link color='inherit' href='#'>
                    Breadcrumb
                </Link>
            </Breadcrumb>
        )

        expect(container).toMatchSnapshot()
    })
})
