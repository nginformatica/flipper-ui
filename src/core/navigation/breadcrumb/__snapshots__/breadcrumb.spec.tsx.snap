// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer should match snapshot 1`] = `
<div>
  <nav
    class="MuiTypography-root MuiBreadcrumbs-root MuiTypography-body1 MuiTypography-colorTextSecondary"
  >
    <ol
      class="MuiBreadcrumbs-ol"
    >
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="#"
        >
          Flipper-ui
        </a>
      </li>
      <li
        aria-hidden="true"
        class="MuiBreadcrumbs-separator"
      >
        /
      </li>
      <li
        class="MuiBreadcrumbs-li"
      >
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorInherit"
          href="#"
        >
          Breadcrumb
        </a>
      </li>
    </ol>
  </nav>
</div>
`;
